### ISSUE TITLE
p2p network protocol documentation

### ISSUE THREAD

**Issue by pmesnier** [EOSIO/eos#181]
Need to add doxygen formatted comments to all of the net plugin headers

**Comment by pmesnier** [comment]
Adding Doxygen style comments to the net_plugin headers.

**Comment by pmesnier** [comment]
Work still in progress. I should be able to commit later tonight or tomorrow.

**Comment by pmesnier** [comment]
still valid, but pushing to milestone 2

**Comment by blockone-syncclient** [comment]
➤ Phil Mesnier commented:

I've been addressing the two big "winnebago" classes in net_plugin.cpp, net_plugin_impl and connection, by breaking the classes into chains of specialization. This means that instead of a single class with 80 methods, I have 8 classes with an average of 10 methods each, where these are all logically related. In the end the final class contains all of methods, but in bite-sized chunks.

I am also documenting the theory of operation - including state transitions.  This work isn't done yet, but I am setting it aside for now. I'll have to complete it later.

**Comment by andriantolie** [comment]
@pmesnier are we still working at this?

**Comment by pmesnier** [comment]
Certainly! It is low priority, but given the multitude of roles filled by the main classes I think it is important enough to pursue.  Now that the mainnet is live I think a prudent course of action is to provide documentation for groups of functions without actually modifying running code. That will come later.

**Comment by andriantolie** [comment]
Great. I won't close this issue then. Thanks for the reply!

**Comment by brianjohnson5972** [comment]
@wanderingbort or @heifner I think the answer to this now is that it won't be done, do you agree?
